src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -3618,6 +3618,36 @@ - (void)test034_TestMTRDeviceHistoricalEvents
     XCTAssertTrue(eventReportsReceived > 0);
 }
 
+- (void)test035_TestMTRDeviceSubscriptionNotEstablishedOverXPC
+{
+    NSString * const MTRDeviceControllerId = @"MTRController";
+    __auto_type remoteController = [MTRDeviceController
+        sharedControllerWithID:MTRDeviceControllerId
+               xpcConnectBlock:^NSXPCConnection * _Nonnull {
+                   return nil;
+               }];
+
+    __auto_type * device = [MTRDevice deviceWithNodeID:kDeviceId deviceController:remoteController];
+    dispatch_queue_t queue = dispatch_get_main_queue();
+
+    // We should not set up a subscription when creating a MTRDevice with a remote controller.
+    XCTestExpectation * subscriptionExpectation = [self expectationWithDescription:@"Subscription has been set up"];
+    subscriptionExpectation.inverted = YES;
+
+    __auto_type * delegate = [[MTRDeviceTestDelegate alloc] init];
+
+    XCTAssertTrue([device _getInternalState] == MTRInternalDeviceStateUnsubscribed);
+
+    delegate.onAttributeDataReceived = ^(NSArray<NSDictionary<NSString *, id> *> * attributeReport) {
+        [subscriptionExpectation fulfill];
+    };
+
+    [device setDelegate:delegate queue:queue];
+    [self waitForExpectations:@[ subscriptionExpectation ] timeout:30];
```

Comment:
So this test will always take 30 seconds to run?  That is pretty painful, in terms of actually running the tests locally....  Why do we need this bit at all, given the internal state check above?

Reply:
i wanted to wait a few seconds before checking the internal state again and also make sure the inverted exception is not called. would 5 secs be ok or you are saying we do not need to wait just check the state after setting delegate?